src/k8s-engine/homestead.js

```diff
@@ -71,23 +71,69 @@ async function ensureHelmChart(state) {
   return false;
 }
 
+/**
+ * Ensure that we can actually make a connection; sometimes the first few get
+ * lost, possibly because the underlying server is not ready yet.
+ * @param {string} namespace The namespace to forward to.
+ * @param {string} endpoint The endpoint in the namespace to forward to.
+ * @param {number} port The port to forward to on the endpoint.
+ * @param {KubeClient} client
+ */
+async function waitForConnection(namespace, endpoint, port, client) {
+  const targetName = `${namespace}/${endpoint}:${port}`;
+  for (; ;) {
+    const listeningPort = await client.getForwardedPort(namespace, endpoint, port);
+    if (!listeningPort) {
+      await new Promise(resolve => setTimeout(resolve, 1000));
```

Comment:
Could we extract this code into a sleep method to improve clarity?

Reply:
I'll do a function-local `const sleep = util.promisify(setTimeout)`, then.